api/agent/agent.go

```diff
@@ -277,8 +274,10 @@ func (a *agent) submit(ctx context.Context, call *call) error {
 
 	statsDequeueAndStart(ctx)
 
-	// pass this error (nil or otherwise) to end directly, to store status, etc
-	err = slot.exec(ctx, call)
+	// Pass this error (nil or otherwise) to end directly, to store status, etc.
+	// Reusing http.Request.Context for slot.exec() since call.Start() updates
+	// this with execution deadline.
```

Comment:
this seems pretty hairy, is there any way to make this more explicit? move https://github.com/fnproject/fn/pull/1021/files#diff-02ea3c2166e8abe7a3c4684f68d78db5L231 to here?